geode-core/src/main/java/org/apache/geode/distributed/internal/InternalConfigurationPersistenceService.java

```diff
@@ -841,8 +842,15 @@ private Configuration readConfiguration(File groupConfigDir) throws IOException
     return configuration;
   }
 
-  private String getDeployedBy() {
-    Subject subject = cache.getSecurityService().getSubject();
+  String getDeployedBy() {
+    Subject subject = null;
+    try {
+      subject = cache.getSecurityService().getSubject();
+    } catch (AuthenticationRequiredException exception) {
+      // ignored. No user logged in for the deployment
```

Comment:
Might want to log this at debug or trace level.

Reply:
done.